src/pages/dataSetsWip/form/useGetCompulsoryDataElementOperandsOptions.ts

```diff
@@ -0,0 +1,189 @@
+            }
+            return params
+        },
+    },
+}
+
+type DataSetElement = {
+    categoryCombo?: {
+        id: string
+        displayName: string
+    }
+    dataElement: {
+        id: string
+        displayName: string
+        categoryCombo: {
+            id: string
+            displayName: string
+        }
+    }
+    id: string
+    displayName: string
+}
+
+type CategoryOptionCombo = {
+    id: string
+    displayName: string
+}
+
+type CategoryComboTruncated = {
+    id: string
+    displayName: string
+    categoryOptionCombos: CategoryOptionCombo[]
+}
+
+type Option = {
+    id: string
+    displayName: string
+    dataElement: {
+        id: string
+        displayName: string
+    }
+    categoryOptionCombo: {
+        id: string
+        displayName: string
+    }
+}
+
+type QueryResponse = {
+    categoryCombos: {
+        categoryCombos: CategoryComboTruncated[]
+    }
+}
+
+const compareCategoryCombosArray = (
+    currentCCs: string[],
+    newCCs: string[]
+): boolean => {
+    if (currentCCs.length !== newCCs.length) {
+        return true
+    }
+
+    const index = 0
+    for (const cc of newCCs) {
+        if (cc !== currentCCs[index]) {
+            return true
+        }
+    }
+    return false
+}
+
+const getRelevantCategoryCombos = (value: DataSetElement[]) => {
+    return value
+        .map(
+            (dse: DataSetElement) =>
+                dse?.categoryCombo?.id ?? dse?.dataElement?.categoryCombo?.id
+        )
+        .sort()
+}
+
+const getOptions = ({
+    categoryCombos,
+    dataSetElements,
+}: {
+    categoryCombos: any
+    dataSetElements: DataSetElement[]
+}): Option[] => {
+    if (
+        !categoryCombos ||
+        !dataSetElements ||
+        !categoryCombos?.length ||
+        !dataSetElements?.length
+    ) {
+        return []
+    }
+    const catComboMap = new Map<string, CategoryCombo[]>(
+        categoryCombos?.map((cc: CategoryCombo) => [
+            cc.id,
+            cc.categoryOptionCombos,
+        ])
+    )
+
+    const options = dataSetElements.flatMap((dse) => {
+        const categoryComboId =
+            dse?.categoryCombo?.id ?? dse.dataElement.categoryCombo.id
+        const categoryOptionCombos = catComboMap.get(categoryComboId)
+        // return categoryOptionCombos
+        return categoryOptionCombos?.map((coc: CategoryOptionCombo) => ({
+            id: `${dse?.dataElement?.id}.${coc?.id}`,
+            displayName: `${dse.displayName}: ${coc.displayName}`,
+            dataElement: {
+                id: dse?.dataElement?.id,
+                displayName: dse?.dataElement?.displayName,
+            },
+            categoryOptionCombo: {
+                id: coc?.id,
+                displayName: coc?.displayName,
+            },
+        }))
+    })
+    return options.filter((opt) => opt !== undefined)
+}
+
+export const useGetCompulsoryDataElementOperandsOptions = () => {
+    const { input: dseInput } = useField('dataSetElements')
```

Comment:
I would prefer to take this as a parameter. This makes it more clear in `CompulsoryDataElementsTransfer` that the field depends on `dataSetElements`